frontend/src/PastEncounterList.tsx

```diff
@@ -0,0 +1,196 @@
+import ButtonGroup from "@material-ui/core/ButtonGroup";
+import * as firebase from "firebase/app";
+
+const useStyles = makeStyles((theme: Theme) => createStyles({
+  root: {
+    width: "100%",
+  },
+  paper: {
+    width: "100%",
+    marginBottom: theme.spacing(2),
+  },
+  table: {
+    minWidth: 750,
+  },
+  visuallyHidden: {
+    border: 0,
+    clip: "rect(0 0 0 0)",
+    height: 1,
+    margin: -1,
+    overflow: "hidden",
+    padding: 0,
+    position: "absolute",
+    top: 20,
+    width: 1,
+  },
+  viewEncounterDialog: {
+    margin: theme.spacing(1)
+  }
+}));
+
+interface PastEncounterListProps {
+  user: firebase.User;
+  refresh: boolean; // Used to force refresh from server.
+}
+
+const headers = [
+  {
+    id: "time",
+    numeric: false,
+    disablePadding: false,
+    label: "Scheduled Time",
+  },
+  { id: "name", numeric: false, disablePadding: false, label: "Patient" },
+  { id: "title", numeric: false, disablePadding: false, label: "Title" },
+  {
+    id: "advocates",
+    numeric: false,
+    disablePadding: false,
+    label: "Advocates",
+  },
+  { id: "status", numeric: false, disablePadding: false, label: "Status" },
+  { id: "actions", numeric: false, disablePadding: false, label: "" },
+];
+
+export default function PastEncounterList({ user, refresh }: PastEncounterListProps) {
+  const classes = useStyles();
+  console.log("PastEncounterList");
+  const [encounters, setEncounters] = React.useState<Array<Encounter>>([])
+  const [viewOpen, setViewOpen] = React.useState(false);
+  const [viewEncounterIndex, setViewEncounterIndex] = React.useState<number>(0);
+  const [transcript, setTranscript] = React.useState(String);
+
+  const refreshEncounters = React.useCallback(() => {
+    console.log("Past Encounter List: refreshEncounters");
+    let listEncounters = firebase.functions().httpsCallable("listEncountersByStatus");
+    listEncounters({ userId: "myuser", status: EncounterState.COMPLETE })
+      .then((response) => {
+        let newEncounters = response.data.map((entry: Encounter) => {
+          console.log(JSON.stringify(entry));
+          return entry;
+        });
+        setEncounters(newEncounters);
+      })
+      .catch((err) => {
+        console.log("ERROR: " + JSON.stringify(err));
+      });
+  }, [setEncounters]);
+
+  const fetchTranscript = React.useCallback((encounterId: String) => {
+    console.log("Getting transcript");
+    let serverFunction = firebase.functions().httpsCallable("getEncounterTranscript");
+    serverFunction({userId: user.uid, encounterId: encounterId}).then(function (response) {
+        console.log("Downloaded transcript to memory");
+        console.log(response.data);
+        let newTranscript = response.data.split ('\n').map((item: React.ReactNode, i: string | number | undefined) => <div key={i}>{item}</div>); 
+        setTranscript(newTranscript);
+    }).catch((err) => {
+        console.log(err);
+    });
+  }, [encounters, transcript, setTranscript]);
```

Comment:
i don't think you need encounters or transcript here as dependencies.

Reply:
Yes, fixed.